internal/factsengine/gatherers/systemd_v2.go

```diff
@@ -0,0 +1,131 @@
+package gatherers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+// nolint:gochecknoglobals
+var (
+	SystemDUnitError = entities.FactGatheringError{
+		Type:    "systemd-unit-error",
+		Message: "error getting systemd unit properties",
+	}
+
+	SystemDDecodingError = entities.FactGatheringError{
+		Type:    "systemd-decoding-error",
+		Message: "error decoding systemd unit status",
+	}
+)
+
+type SystemDGathererV2 struct {
```

Comment:
A question about this: should we always define a new struct even when it's the same as the previous version? maybe a type alias makes more sense in such case?

Reply:
A new struct is needed. We need to implement the `Gather` function for that.
We could define the new struct as `type SystemDGathererV2 SystemDGatherer`, but I prefer to keep it explicit, to understand the code easier.
I prefer some little duplication in this case